src/target/riscv/riscv-013.c

```diff
@@ -1689,7 +1900,103 @@ static int read_memory(struct target *target, target_addr_t address,
 	return result;
 }
 
-static int write_memory(struct target *target, target_addr_t address,
+static int read_memory(struct target *target, target_addr_t address,
+		uint32_t size, uint32_t count, uint8_t *buffer)
+{
+	RISCV013_INFO(info);
+	if (info->progbufsize >= 2) {
+		return read_memory_progbuf(target, address, size, count, buffer);
+	} else if ((get_field(info->sbcs, DMI_SBCS_SBVERSION) == 1) && (
```

Comment:
would it be useful to provide some sort of command or interface that would prioritize system bus accesses over progbuf accesses? I agree in general progbuf is better/simpler, but maybe there are certain places (e.g. hart not halted) that you want to just use System Bus.

Reply:
Maybe. Without a real use case I'm hesitant to add a config option that is one more thing to test and verify what setting is used in bug reports.
If you think it's important, can you file an issue? I feel this pull request has had too much feature creep already.